piptools/utils.py

```diff
@@ -366,10 +370,17 @@ def get_compile_command(click_ctx):
                 left_args.append(shlex_quote(arg))
             # Append to args the option with a value
             else:
-                left_args.append(
-                    "{option}={value}".format(
-                        option=option_long_name, value=shlex_quote(force_text(val))
+                if option.name == "pip_args":
+                    left_args.append(
+                        "{option}={value}".format(
+                            option=option_long_name, value=repr(fs_str(force_text(val)))
+                        )
+                    )
```

Comment:
Could you add a comment about why `pip_args` handled in that way?

Reply:
Will push soon:
```python
if option.name == "pip_args":
    # shlex_quote would produce functional but noisily quoted results,
    # e.g. --pip-args='--cache-dir='"'"'/tmp/with spaces'"'"''
    # Instead, we try to get more legible quoting via repr:
```

Reply:
Neat!